tuf/api/metadata.py

```diff
@@ -134,7 +134,10 @@ def __init__(
     ):
         self.signed: T = signed
         self.signatures = signatures if signatures is not None else {}
-        self.unrecognized_fields: Dict[str, Any] = unrecognized_fields or {}
+        if unrecognized_fields is None:
+            unrecognized_fields = {}
+
+        self.unrecognized_fields = unrecognized_fields
```

Comment:
I think it would be better if you add some information about the expected type of `unrecognized_fields` like:
```suggestion
        self.unrecognized_fields: Dict[str, Any] = unrecognized_fields
```
That applies to all places where you have used `self.unrecognized_fields = unrecognized_fields`

Reply:
I wouldn't object to that but I'm not going to require it either: static checkers will correctly deduce the type as `Dict[str, Any]` here (because of the original argument type) and that's enough for me.